core/src/main/java/com/yahoo/imapnio/async/request/UidFetchCommand.java

```diff
@@ -16,7 +18,19 @@ public class UidFetchCommand extends AbstractFetchCommand {
      * @param items the data items
      */
     public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final String items) {
-        super(true, msgsets, items);
+        this(msgsets, items, null);
+    }
+
+    /**
+     * Initializes a {@link UidFetchCommand} with the {@link MessageNumberSet} array.
+     *
+     * @param msgsets the set of message set
+     * @param items the data items
+     * @param partialExtUidFetchInfo partial extension uid fetch info
+     */
+    public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final String items,
```

Comment:
not critical but a pattern. Let's not let user to pass in @Nullable partialExtUidFetchInfo, a nullable value makes parameter list looks confusing (required or not required? such). Change this to @Nonnull instead, and let line 21 to call super(...)